### PR TITLE
[RDY] Macro cleanup (2)

### PR DESCRIPTION
**Pull request by ghost**
Regarding the removal of `UNICODE16`, see https://github.com/neovim/neovim/pull/568#issuecomment-40870749.

### CODE REVIEW

**Comment by Pyrohh** [comment]
bump

File: src/nvim/screen.c [hunk 1]
```diff
@@ -3051,30 +3046,14 @@ win_line (
...
-             * Illegal UTF-8 byte: display as <xx>.
-             * Non-BMP character : display as ? or fullwidth ?.
-             */
-# ifdef UNICODE16
-            if (mb_c < 0x10000)
-# endif
+              || (mb_l > 1 && (!vim_isprintc(mb_c)))) {
+            // Illegal UTF-8 byte: display as <xx>.
+            // Non-BMP character : display as ? or fullwidth ?.
             {
```
Comment (justinmk) [inline src/nvim/screen.c]: the double `{{` here can probably be removed now

File: src/nvim/globals.h [hunk 2]
```diff
@@ -838,13 +834,6 @@ EXTERN int ctrl_x_mode INIT(= 0);       /* Which Ctrl-X mode are we in? */
 
 EXTERN int no_abbr INIT(= TRUE);        /* TRUE when no abbreviations loaded */
 
-#ifdef USE_EXE_NAME
-EXTERN char_u   *exe_name;              /* the name of the executable */
```
Comment (justinmk) [inline src/nvim/globals.h]: `:help $VIM`, item 3, reveals that this exists to support running vim from any old directory, and it walks up from the exe location to look for a user `.vimrc` file.

Regardless if we want to support this, the platform-specific nature of this define is unnecessary. We can easily add support for this feature later if needed.

File: src/nvim/macros.h [hunk 3]
```diff
@@ -97,15 +97,9 @@
  */
 #define vim_isbreak(c) (breakat_flags[(char_u)(c)])
 
-#ifdef BINARY_FILE_IO
-# define WRITEBIN   "wb"        /* no CR-LF translation */
-# define READBIN    "rb"
-# define APPENDBIN  "ab"
```
Comment (justinmk) [inline src/nvim/macros.h]: Maybe move this to win_defs.h? cc @equalsraf

Comment (justinmk) [inline src/nvim/macros.h]: @equalsraf Added this to his win32 PR, so let's not remove this block.

Comment (Pyrohh) [inline src/nvim/macros.h]: :+1:

File: src/nvim/screen.c (hunk 1 above)
Comment (Pyrohh) [inline src/nvim/screen.c]: good catch

**Comment by Pyrohh** [comment]
Made the recommended changes and re-pushed.

**Comment by Pyrohh** [comment]
Unless you have any more advice @justinmk, I'll mark this as ready.

**Comment by justinmk** [comment]
merged. nice cleanup!

**Comment by Pyrohh** [comment]
thanks! Why the merge commit though? I don't care much, just curious.

**Comment by justinmk** [comment]
I use `git merge --no-ff` so that a group of commits is "grouped" in the git graph. 

See discussions:

- https://github.com/neovim/neovim/pull/703#issuecomment-42776378
- https://github.com/neovim/neovim/pull/710#issuecomment-43458549

For single-commit PRs, I don't use a merge commit because there's nothing to group. If it's rebased onto master, I'll commit it with no modifications. If it's behind master, I'll rebase it and amend the commmit message with the PR number.

**Comment by Pyrohh** [comment]
Thanks, the existence of that merge commit makes a lot more sense when viewed with `git log --graph`.

**Pull request closed without merging**